api/enki.py

```diff
@@ -0,0 +1,633 @@
+                # Enki told us about a book we didn't ask
+                # for. This shouldn't happen, but if it does we should
+                # do nothing further.
+                continue
+            seen_identifiers.add(identifier.identifier)
+            result = self.set_metadata(identifier, metadata)
+            if not isinstance(result, CoverageFailure):
+                result = self.handle_success(identifier)
+            batch_results.append(result)
+
+        # Create a CoverageFailure object for each original identifier
+        # not mentioned in the results.
+        for identifier_string in identifier_strings:
+            if identifier_string not in seen_identifiers:
+                identifier, ignore = Identifier.for_foreign_id(
+                    self._db, Identifier.ENKI_ID, identifier_string
+                )
+                result = CoverageFailure(
+                    identifier, "Book not in collection", data_source=self.output_source, transient=False
+                )
+                batch_results.append(result)
+        return batch_results
+
+    def handle_success(self, identifier):
+        return self.set_presentation_ready(identifier)
+
+    def process_item(self, identifier):
+        results = self.process_batch([identifier])
+        return results[0]
+
+class BibliographicParser(object):
+
+    """Helper function to parse JSON"""
+    def process_all(self, json_data, xpath, namespaces=None, handler=None, parser=None):
+        data = json.loads(json_data)
+	returned_titles = data["result"]["titles"]
+	titles = returned_titles
+	for book in returned_titles:
+            print "A book titled '%s'" % book["title"]
+            print book
+            print "\n"
+	    data = self.process_one(book, namespaces)
+            if data:
+                yield data
+
+    DELIVERY_DATA_FOR_AXIS_FORMAT = {
+        "Blio" : None,
+        "Acoustik" : None,
+        "ePub" : (Representation.EPUB_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM),
+        "PDF" : (Representation.PDF_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM),
+    }
+
+    log = logging.getLogger("Enki Bibliographic Parser")
+
+    @classmethod
+    def parse_list(self, l):
+        """Turn strings like this into lists:
+
+        FICTION / Thrillers; FICTION / Suspense; FICTION / General
+        Ursu, Anne ; Fortune, Eric (ILT)
+        """
+        return [x.strip() for x in l.split(";")]
+
+    def __init__(self, include_availability=True, include_bibliographic=True):
+        self.include_availability = include_availability
+        self.include_bibliographic = include_bibliographic
+
+    def extract_availability(self, circulation_data, element, ns):
+	primary_identifier = IdentifierData(Identifier.ENKI_ID, element["id"])
+        if not circulation_data:
+            circulation_data = CirculationData(
+                data_source=DataSource.ENKI,
+                primary_identifier=primary_identifier,
+            )
```

Comment:
When you create a `CirculationData` object you need to include `FormatData` objects that make it clear what media types/DRM schemes the book is actually available in. Otherwise it will be treated as a book that's licensed but not available in any format. 

The last time I dealt with Enki it seemed like all books were in EPUB format with the Adobe DRM scheme, but I don't know if that's always the case.